src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,115 @@
+package seedu.duke;
+
+/**
+ * Parser processes each command from Ui and executes program accordingly.
+ */
+public class Parser {
+    private TaskList taskLists;
+
+    /**
+     * Initializes an instance of a Parser.
+     *
+     * @param taskLists List of Tasks stored in a TaskList class.
+     */
+    public Parser(TaskList taskLists) {
+        this.taskLists = taskLists;
+    }
+
+    /**
+     * Checks for invalid inputs.
+     *
+     * @param input String that is input by user.
+     * @throws DukeException if input does not contain any of the keywords in the method.
+     */
+    public void checkForInvalidInput(String input) throws DukeException {
```

Comment:
Lol this is very deep nesting, try to refactor it into multiple &&. Refer to: https://nus-cs2103-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting.